airflow/models/dagbag.py

```diff
@@ -398,6 +400,9 @@ def _process_modules(self, filepath, mods, file_last_changed_on_disk):
             dag.fileloc = mod.__file__
             try:
                 dag.timetable.validate()
+                # create a copy of params before validating
```

Comment:
Why though? We should add that to this comment or remove it.

Reply:
We can actually validate directly without copy since we are not updating the params. Will update it